### PR TITLE
Adding the possibility to search for the blocks to insert

### PR DESCRIPTION
**Pull request by youknowriad**
refs #82 

This PR adds the possibility to filter the blocks menu using the search input. It also adds some scrolling to the blocks menu.

kapture 2017-02-17 at 11 43 11

### CODE REVIEW

**Comment by jasmussen** [comment]
😍

Works great, thanks for doing this!

Good to merge for now, but the behavior can be finetuned further. It should be keyboard shortcuttable all the way through, as a dogma:

screen shot 2017-02-17 at 12 34 16

So, as soon as you invoke the inserter, you can use the arrow keys to pick a block and press enter to insert. So we can't set focus on the search field automatically — my bad, I added this to the prototype. 

In addition to picking blocks using the arrow keys, you can start typing — even if the search field isn't focused — `ima`and you are filtered down to Image, with the focus rectangle now around the image. Press enter to insert the image. 

To put it simply, if you're a keyboard shortcut person, you should be able to:

1. Type `/ on a new line to insert a block (see #24)
2. Type `ima` to filter down to just the Image
3. Press enter to insert the image block.

I'll open this as a separate ticket. But nice work so far!

**Comment by mtias** [comment]
Thanks! Let's 🚢  and refine.

**Comment by annaephox** [comment]
This looks really good! Quick question (maybe already thought of :-) : are you thinking of indexing on fields other than the block name/descriptor and HTML code/id? As the [number of blocks](https://docs.google.com/spreadsheets/d/1r4Rfj4D9CyFkxQr0JFi3iVKkRte154M8BkDpkMPysp4/edit#gid=0) grows (potentially to a v large number), would it make sense to add block-type meta data to what is searchable?
Example: Show me all "media" blocks? Show me all of "my custom" blocks? Show me all "new" blocks?

**Comment by youknowriad** [comment]
Tks @annaephox for the feedback. Actually, I haven't thought about it, but it could be a good quick-win feature.

**Comment by mtias** [comment]
@annaephox yes, makes sense. I think we should add to the prototype a lot of dummy blocks just to see how it scales and feels, and whether we can quickly sort through blocks. That would cast light on where it falls short.
